src/service/viewer-impl.js

```diff
@@ -257,6 +267,22 @@ export class Viewer {
     /** @const @private {!Promise<boolean>} */
     this.isTrustedViewer_ = trustedViewerPromise;
 
+    /** @const @private {!Promise<string>} */
+    this.viewerOrigin_ = new Promise(resolve => {
+      if (!this.isEmbedded()) {
+        // Viewer is only determined for iframed documents at this time.
+        resolve('');
+      } else if (this.win.location.ancestorOrigins &&
+          this.win.location.ancestorOrigins.length > 0) {
+        resolve(this.win.location.ancestorOrigins[0]);
+      } else {
+        // Race to resolve with a timer.
+        timer.delay(() => { resolve(''); }, VIEWER_ORIGIN_TIMEOUT_);
```

Comment:
Just this will suffice:
`timer.delay(() => resolve(''), VIEWER_ORIGIN_TIMEOUT_);`